clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -22933,6 +22933,35 @@ llvm::CallInst *CodeGenFunction::EmitFPBuiltinIndirectCall(
             .Case("sincos", llvm::Intrinsic::fpbuiltin_sincos)
             .Case("exp10", llvm::Intrinsic::fpbuiltin_exp10)
             .Case("rsqrt", llvm::Intrinsic::fpbuiltin_rsqrt)
+            .Case("__spirv_ocl_sin", llvm::Intrinsic::fpbuiltin_sin)
+            .Case("__spirv_ocl_asin", llvm::Intrinsic::fpbuiltin_asin)
+            .Case("__spirv_ocl_sinh", llvm::Intrinsic::fpbuiltin_sinh)
+            .Case("__spirv_ocl_asinh", llvm::Intrinsic::fpbuiltin_asinh)
+            .Case("__spirv_ocl_cos", llvm::Intrinsic::fpbuiltin_cos)
+            .Case("__spirv_ocl_acos", llvm::Intrinsic::fpbuiltin_acos)
+            .Case("__spirv_ocl_cosh", llvm::Intrinsic::fpbuiltin_cosh)
+            .Case("__spirv_ocl_acosh", llvm::Intrinsic::fpbuiltin_acosh)
+            .Case("__spirv_ocl_tan", llvm::Intrinsic::fpbuiltin_tan)
+            .Case("__spirv_ocl_tanh", llvm::Intrinsic::fpbuiltin_tanh)
+            .Case("__spirv_ocl_atan", llvm::Intrinsic::fpbuiltin_atan)
+            .Case("__spirv_ocl_atan2", llvm::Intrinsic::fpbuiltin_atan2)
+            .Case("__spirv_ocl_atanh", llvm::Intrinsic::fpbuiltin_atanh)
+            .Case("__spirv_ocl_sincos", llvm::Intrinsic::fpbuiltin_sincos)
+            .Case("__spirv_ocl_pow", llvm::Intrinsic::fpbuiltin_pow)
+            .Case("__spirv_ocl_ldexp", llvm::Intrinsic::fpbuiltin_ldexp)
+            .Case("__spirv_ocl_exp", llvm::Intrinsic::fpbuiltin_exp)
+            .Case("__spirv_ocl_exp2", llvm::Intrinsic::fpbuiltin_exp2)
+            .Case("__spirv_ocl_exp10", llvm::Intrinsic::fpbuiltin_exp10)
+            .Case("__spirv_ocl_expm1", llvm::Intrinsic::fpbuiltin_expm1)
+            .Case("__spirv_ocl_hypot", llvm::Intrinsic::fpbuiltin_hypot)
+            .Case("__spirv_ocl_log", llvm::Intrinsic::fpbuiltin_log)
+            .Case("__spirv_ocl_log2", llvm::Intrinsic::fpbuiltin_log2)
+            .Case("__spirv_ocl_log10", llvm::Intrinsic::fpbuiltin_log10)
+            .Case("__spirv_ocl_log1p", llvm::Intrinsic::fpbuiltin_log1p)
+            .Case("__spirv_ocl_sqrt", llvm::Intrinsic::fpbuiltin_sqrt)
+            .Case("__spirv_ocl_rsqrt", llvm::Intrinsic::fpbuiltin_rsqrt)
+            .Case("__spirv_ocl_erf", llvm::Intrinsic::fpbuiltin_erf)
+            .Case("__spirv_ocl_erfc", llvm::Intrinsic::fpbuiltin_erfc)
```

Comment:
Could the SYCL implementation be using other built-in functions to achieve this without needing these special rules? E.g. could we be calling another `sqrt` function than `__spirv_ocl_sqrt` and get this behavior directly without needing these new cases?

Reply:
If we want to avoid adding these new cases, then we need to have the FE generates different names (same names than std::* functions?) for the sycl math functions.
These sycl math functions are all defined in SPRIVBuiltins.td and tablegen generates those __spirv_ocl_*. But not sure if we can prefix them differently. @Andy Kaylor what do you think? 
Would you want that for all math functions or only specific ones?